src/plugins/workspace/public/components/workspace_app.tsx

```diff
@@ -1,18 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
 import React, { useEffect } from 'react';
 import { EuiPage, EuiPageBody } from '@elastic/eui';
 import { I18nProvider } from '@osd/i18n/react';
 import { matchPath, Route, Switch, useLocation } from 'react-router-dom';
 
 import { ROUTES } from './routes';
 import { useOpenSearchDashboards } from '../../../opensearch_dashboards_react/public';
-import { ChromeBreadcrumb } from '../../../../core/public';
+import { ChromeBreadcrumb, PublicAppInfo } from '../../../../core/public';
 import { WORKSPACE_APP_NAME } from '../../common/constants';
+import { WorkspaceTemplate } from '../../../../core/types';
 
 export const WorkspaceApp = ({ appBasePath }: { appBasePath: string }) => {
   const {
-    services: { chrome },
+    services: { chrome, application },
   } = useOpenSearchDashboards();
   const location = useLocation();
+  let workspaceTemplates = [] as WorkspaceTemplate[];
+  const templateFeatureMap = new Map<string, PublicAppInfo[]>();
+  application?.applications$.subscribe((map) =>
+    map.forEach((app) => {
+      const { workspaceTemplate: templates = [] } = app;
+      workspaceTemplates.push(...templates);
+      for (const template of templates) {
+        const features = templateFeatureMap.get(template.id) || [];
+        features.push(app);
+        templateFeatureMap.set(template.id, features);
+      }
+    })
+  );
+
+  workspaceTemplates = [...new Set(workspaceTemplates)];
```

Comment:
Seems this `workspaceTemplates` wasn't be used in the workspace, is it only a demo here? Do we need sort by order here? How about separate a hooks to do this?

Reply:
thanks for the comment.Yes, for demo only. Let me extract this logic into a hook.